### PR TITLE
Upgrade NodeJS to version 16.x LTS and remove deprecated apt-key usage

### PR DESCRIPTION
**Pull request by kissifrot**
Switch to current LTS version of NodeJS and use keyrings files directly instead of now deprecated apt-key usage

### CODE REVIEW

File: infrastructure/docker/services/builder/Dockerfile [hunk 1]
```diff
@@ -2,14 +2,13 @@ ARG PROJECT_NAME
 
 FROM ${PROJECT_NAME}_php-base
 
-ARG NODEJS_VERSION=14.x
-RUN echo "deb https://deb.nodesource.com/node_${NODEJS_VERSION} bullseye main" > /etc/apt/sources.list.d/nodejs.list \
-    && apt-key adv --fetch-keys https://deb.nodesource.com/gpgkey/nodesource.gpg.key
+ARG NODEJS_VERSION=16.x
+RUN curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | tee /usr/share/keyrings/nodesource.gpg >/dev/null \
```
Comment (lyrixx) [inline infrastructure/docker/services/builder/Dockerfile:6]: I think this could be simpler : 

```
curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key| gpg --dearmor  > /usr/share/keyrings/nodesource.gpg
```

---

I have another question, why did you change the way the package signature is checked?

Comment (kissifrot) [reply infrastructure/docker/services/builder/Dockerfile:6]: I will simplify the command, good catch (I followed the way its done by NodeJS scripts)

apt-key will be removed in future distributions versions (see https://wiki.debian.org/DebianRepository/UseThirdParty)

Comment (lyrixx) [reply infrastructure/docker/services/builder/Dockerfile:6]: Oh, thanks for this Information. I didn't know.

Comment (kissifrot) [reply infrastructure/docker/services/builder/Dockerfile]: Simplified commands 🙂

**Review by lyrixx — APPROVED** [review]
👍🏼

**Comment by lyrixx** [comment]
Thanks
